hub/src/server/server.ts

```diff
@@ -213,6 +215,42 @@ export class HubServer {
       }
     }
 
+    const ifMatchTag = requestHeaders['if-match']
+    const ifNoneMatchTag = requestHeaders['if-none-match']
+
+    // only one match-tag header should be set
+    if (ifMatchTag && ifNoneMatchTag) {
+      throw new PreconditionFailedError('Request should not contain both if-match and if-none-match headers')
+    }
+
+    if (!this.driver.supportsETagMatching) {
+      // allow overwrites if tag is wildcard 
+      if (ifMatchTag && ifMatchTag !== '*') {
```

Comment:
Should we have an explicit `ifMatchTag === '*' ... then overwrite` condition? 
Looks like the overwrite scenario may fall through to what I'm considering "legacy" behavior, but IMO we should have an explicit check

Reply:
How would you implement this? What would be in the body of the `ifMatchTag === '*'` conditional? 

I thought about doing this, but wasn't sure how to do it without repeating any code. Since the (over)write code happens below, I figured I'd simply let wildcard tags pass through via the conditional

Reply:
Hmm, you're right, looks like the body would be a noop, or oddly duplicated write code..
Perhaps the condition would be more like 
```js
if (!ifMatchTag && !ifNoneMatchTag) { 
  log('deprecated/legacy write request')
}
```

This is in anticipation of a future where that `log` is replaced with an error, ensuring future Gaia client implementations don't end up with the same concurrency bugs as (current) blockstack.js. 

I don't feel that strongly about this -- perhaps there are good arguments for why this condition should be indefinitely allowed? @kantai @jcnelson

Reply:
The way the condition is checked now works for me. I'd be against adding that log entry for now, since it would generate a _lot_ of entries. Plus, this could be logged on an HTTP front-end.